drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scesim/backend/server/SceSimServiceImpl.java

```diff
@@ -0,0 +1,310 @@
+
+    @Inject
+    private MVELEvaluator evaluator;
+
+    @Inject
+    private SaveAndRenameServiceImpl<String, Metadata> saveAndRenameService;
+
+    private SafeSessionInfo safeSessionInfo;
+
+    public SceSimServiceImpl() {
+    }
+
+    @Inject
+    public SceSimServiceImpl(final SessionInfo sessionInfo) {
+        safeSessionInfo = new SafeSessionInfo(sessionInfo);
+    }
+
+    @PostConstruct
+    public void init() {
+        saveAndRenameService.init(this);
+    }
+
+    @Override
+    public Path create(final Path context,
+                       final String fileName,
+                       final String content,
+                       final String comment) {
+        try {
+            final org.uberfire.java.nio.file.Path nioPath = Paths.convert(context).resolve(fileName);
+            final Path newPath = Paths.convert(nioPath);
+
+            if (ioService.exists(nioPath)) {
+                throw new FileAlreadyExistsException(nioPath.toString());
+            }
+
+            ioService.write(nioPath,
+                            content,
+                            commentedOptionFactory.makeCommentedOption(comment));
+
+            return newPath;
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public String load(final Path path) {
+        try {
+            final String content = ioService.readAllString(Paths.convert(path));
+
+            return content;
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
+        }
+    }
+
+    @Override
+    public SceSimModelContent loadContent(final Path path) {
+        return super.loadContent(path);
+    }
+
+    @Override
+    protected SceSimModelContent constructContent(Path path,
+                                                  Overview overview) {
+        //Signal opening to interested parties
+        resourceOpenedEvent.fire(new ResourceOpenedEvent(path,
+                                                         safeSessionInfo));
+
+        return new SceSimModelContent(new SceSimModel(load(path)),
+                                      overview);
+    }
+
+    @Override
+    public Path save(final Path resource,
+                     final String content,
+                     final Metadata metadata,
+                     final String comment) {
+        try {
+            Metadata currentMetadata = metadataService.getMetadata(resource);
+            ioService.write(Paths.convert(resource),
+                            content,
+                            metadataService.setUpAttributes(resource,
+                                                            metadata),
+                            commentedOptionFactory.makeCommentedOption(comment));
+
+            //Invalidate Package-level DMO cache as SceSims have changed.
+            invalidateDMOPackageCache.fire(new InvalidateDMOPackageCacheEvent(resource));
```

Comment:
Presumably this was copied from the existing Enumeration editor? 

There's probably no need to invalidate the DMO cache when a Simulation is saved?